spec/folio_client/authenticator_spec.rb

```diff
@@ -7,49 +7,69 @@
+    stub_request(:post, "#{url}/authn/login").to_return(status: http_status, body: http_body)
+  end
+
+  describe ".token" do
+    let(:instance) { described_class.new(login_params, connection) }
 
-  context "when correct credentials" do
     before do
-      stub_request(:post, "#{url}/authn/login")
-        .to_return(status: 200, body: "{\"okapiToken\" : \"#{token}\"}")
+      allow(described_class).to receive(:new).and_return(instance)
+      allow(instance).to receive(:token)
     end
 
-    describe ".token" do
-      let(:instance) do
-        described_class.new(login_params, connection)
-      end
+    it "invokes #token on a new instance" do
+      described_class.token(login_params, connection)
+      expect(instance).to have_received(:token).once
+    end
+  end
+
+  describe "#token" do
+    subject(:authenticator) { described_class.new(login_params, connection) }
 
-      before do
-        allow(described_class).to receive(:new).and_return(instance)
-        allow(instance).to receive(:token)
+    context "when correct credentials" do
+      it "parses the token from the response" do
+        expect(authenticator.token).to eq(token)
       end
+    end
 
-      it "invokes #token on a new instance" do
-        described_class.token(login_params, connection)
-        expect(instance).to have_received(:token).once
+    context "when incorrect credentials" do
+      let(:http_status) { 422 }
+      let(:http_body) { "{\"error\" : \"get bent\"}" }
+
+      it "raises an unauthorized error" do
+        expect { authenticator.token }.to raise_error(FolioClient::UnauthorizedError)
       end
     end
 
-    describe "#token" do
-      subject(:authenticator) { described_class.new(login_params, connection) }
+    context "when client lacks privileges" do
+      let(:http_status) { 403 }
+      let(:http_body) { "{\"error\" : \"get bent\"}" }
 
-      it "parses the token from the response" do
-        expect(authenticator.token).to eq(token)
+      it "raises a forbidden error" do
+        expect { authenticator.token }.to raise_error(FolioClient::ForbiddenError)
       end
     end
-  end
 
-  context "when incorrect credentials" do
-    before do
-      stub_request(:post, "#{url}/authn/login")
-        .to_return(status: 422, body: "{\"errror\" : \"get bent\"}")
+    context "when service is unavailable" do
+      let(:http_status) { 500 }
+      let(:http_body) { "{\"error\" : \"get bent\"}" }
+
+      it "raises a service unavailable exception" do
+        expect { authenticator.token }.to raise_error(FolioClient::ServiceUnavailable)
+      end
     end
 
-    describe "#token" do
-      subject(:authenticator) { described_class.new(login_params, connection) }
+    context "when the truly unexpected happens" do
+      let(:http_status) { 666 }
+      let(:http_body) { "{\"error\" : \"huh?\"}" }
 
-      it "raises the correct exception" do
-        expect { authenticator.token }.to raise_error(FolioClient::UnauthorizedError)
```

Comment:
I think @justinlittman specifically asked @lwrubel NOT to use StandardError so it was easier to trap for FolioClient errors ...